### PR TITLE
fix: noop not published for subscription topics

### PR DESCRIPTION
**Pull request by chris13524**
# Description

Fixes that noop (4050) message to extend topic TTL from 5m to 30d was not being published for subscriptions. The previous approach was intentional, as publishing to every subscription topic will take a long time (calculated 30-50 minutes for current number of topics) and additionally the noop message was already being published whenever the subscription was created for the first time.

However, while the noop message was already published when a subscription was created, this only increased the topic TTL to 30 days temporarily and after 30 days the topic was back to having only a TTL of 5 minutes. The topic still needs messages flowing over it for the topic to be renewed to 30 days.

[Slack conversation](https://walletconnect.slack.com/archives/C03SMNKLPU0/p1707142307187359)

## How Has This Been Tested?

Not tested

## Due Diligence

* [ ] Breaking change
* [ ] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

**Comment by chris13524** [comment]
This isn't going to work. The initial subscribe will take 50m which will be much longer than the health check allowance since the websocket client is created prior to the HTTP server starting. Also, the subscriptions themselves will be lost after 5 minutes so presumably won't be renewed anyway.

File: src/services/websocket_server/mod.rs [hunk 1]
```diff
@@ -200,18 +198,45 @@ async fn resubscribe(
...
+        async move {
+            let client = &client;
+            let relay_http_client = &relay_http_client;
+            let metrics = metrics.as_ref();
+            let start = Instant::now();
+            let result = futures_util::stream::iter(topics)
+                .map(|topic| async move {
+                    // Subscribe a second time as the initial subscription above may have expired
+                    subscribe_relay_topic(client, &topic, metrics)
+                        .map_ok(|_| ())
```
Comment (xDarksome) [inline src/services/websocket_server/mod.rs:216]: Consider using `drop` in such cases
```rust
.map_ok(drop)
```

**Review by xDarksome — APPROVED** [review]
I would also add some retries to tasks inside the batch, otherwise a single fluke may fail the whole pipeline.

**Comment by chris13524** [comment]
> I would also add some retries to tasks inside the batch, otherwise a single fluke may fail the whole pipeline.

@xDarksome thanks, yeah those relay functions are already being retired 10 times so we should be good

**Pull request merged**